guide/popular-topics/errors.md

```diff
@@ -0,0 +1,155 @@
+
+```
+GET    - Used to retrieve a piece of data
+POST   - Used to send a piece of data
+PATCH  - Used to modify a piece of data
+PUT    - Used to completely replace a piece of data
+DELETE - Used to completely delete a piece of data
+```
+
+In this particular example we can see we are trying to access a piece of data, specifically, a message.
+
+## Common Discord.js and API errors
+
+### An invalid token was provided.
+
+This is a very common error, it originates from a wrong token being passed into `client.login()`. The most common causes of this error are:
+
+- Not importing the config or env file correctly
+- Copying the client secret instead of the bot token (token is alphanumerical and 3 parts delimited by a period while client secret is all numbers)
+- Simply showing the token and copying that, instead of clicking regenerate and copying that.
+
+<branch version="12.x">
+
+::: warning
+On master there used to be an issue where the token was not prefixed correctly which resulted in valid tokens being marked as invalid. If you have verified that all of the above is not the case, make sure you have updated discord.js to the latest commit.
+:::
+
+</branch>
+
+### Request to use token, but token was unavailable to the client.
+
+Another common error, this error originates from the client attempting to execute an action which requires the token but the token not being available. This is most commonly caused by destroying the client and then attempting to execute an action.
+
+<branch version="12.x">
+
+### MessageEmbed field names may not be empty.
+
+This error originates from attempting to call `MessageEmbed.addField()` without the first parameter, which is a title. If you would like the title to be empty for a reason, you should use a zero width space, which can be inputted as `\u200b`.
+
+### MessageEmbed field values may not be empty.
+
+This error, in conjunction to the previous error, is the result of calling `MessageEmbed.addField()` without the second parameter, the value. You can use a zero width space if you would like this empty.
+
+</branch>
+
+### The messages must be an Array, Collection, or number.
+
+This error originates from an invalid call to `bulkDelete()`, make sure you are inputting a valid Array or Collection of messages, or a valid number.
+
+### Members didnt't arrive in time.
+
+Another common error, this error originates from the client requesting members from the API through the websocket, and the member chunks not arriving in time and triggering the timeout. The most common cause to this error is a bad connection, however, it can also be caused by a very large amount of members being fetched, upwards of 50 thousand. To fix this, run the bot on a location with better internet, such as a VPS. If this does not work for you, you will have to manually change the member fetching timeout.
```

Comment:
Just fyi: The timeout, at least in master, is hard coded in. Editing would mean patching one line from internals; definitely not a hard task, but for beginners it sure isn't the simplest too.

Reply:
yeah, i know about that, i forgot to mention its hardcoded :^), one of the things i dont really get, in my opinion it should be able to be easily edited through client options or somewhere else, but thats another topic to discuss.

Reply:
Feel free to make it a suggestion on the main repo if you want!

Reply:
yeah, i feel like this will be helpful for people with a large amount of guilds, i will submit a PR/issue to add this later, when i have some time

Reply:
do you happen to know where this interval is hardcoded?

Reply:
This is where the timeout is created

https://github.com/discordjs/discord.js/blob/50ed3293a57aeb0f9c094f00d49ed9874dc5d87a/src/stores/GuildMemberStore.js#L221-L224

I recommend creating a issue truthfully, since a better approach would be refreshing the timeout with every packet we get; but feel free to submit a PR too!

Reply:
Alright, thank you